### PR TITLE
Eventbrite clone

### PR DESCRIPTION
**Pull request by Audrey-Ella-xo**
In this project, we did the following:

- We created the Users, Events and user attended events models and controllers

- We set up and sign in users

- Users can create basic events 

- They can also attend events

- As extra credit, we enabled users to invite other users to attend their created events and only invited users can attend the events.

- We added unit and integration tests 

Thank you for your kind review. We await your speedy response

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  belongs_to :creator, class_name: 'User', foreign_key: 'user_id'
+
+  has_many :user_attended_events
+  has_many :attendees, through: :user_attended_events, source: :user
+
+  has_many :invitations
+  has_many :invitees, through: :invitations, source: :user
+
+  scope :past, -> { where('date < ?', Time.now) }
+  scope :upcoming, -> { !past }
```
Comment (wathigo) [inline app/models/event.rb]: The logic here is not correct.
Both past events and upcoming events are displayed as upcoming events.
See this :point_down: 
Screenshot from 2020-02-19 01-49-54
Let's fix this by using `scope :past, -> { where('date > ?', Time.now) }`

File: app/views/events/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+<div>
+  <div>
+    <%= form_for @event, url: {action: "create"} do |f| %>
+
+      <%= f.label :description %>
+      <%= f.text_field :description %>
+      
+      <%= f.submit "Create my event" %>
+    <% end %>
```
Comment (wathigo) [inline app/views/events/new.html.erb]: Let's update the form to take both event date and venue

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<% if logged_in? %>
+<h2>Description: <%= @event.description %></h2>
+<% else %>
+<h1>Description: <%= @event.description %></h1>
+<% end %>
```
Comment (wathigo) [inline app/views/events/show.html.erb]: Lets display the event date and venue

**Review by wathigo — CHANGES_REQUESTED** [review]
## Changes Required 
### Hello @Audrey-Ella-xo  and @euqueme :wave: 
Great test coverage!
I can see the effort you guys are putting into this project.

**[REQUIRED]**
We have a bug we need to fix.
See the comments :point_down: 

**After implementing the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

File: app/models/event.rb (hunk 1 above)
Comment (Audrey-Ella-xo) [reply app/models/event.rb]: Thanks for noticing this. We fixed the bug, but with a different method

File: app/views/events/show.html.erb (hunk 3 above)
Comment (Audrey-Ella-xo) [reply app/views/events/show.html.erb]: Done

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 🤓 🎉 🎊 

Hi @Audrey-Ella-xo and @euqueme 👋 

You guys successfully implemented changes. May now move on. 😄 

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**